test/basics.js

```diff
@@ -0,0 +1,99 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: loopback-connector-cos
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+require('./init.js');
+var should = require('should');
+var ds, COSModel;
+
+
+describe('cos model creation', function() {
+  before(function(done){
+    ds = global.getDataSource(); //.settings is your config
+
+    COSModel = ds.define('COSModel', {
+      name: {type: String},
+      age: {type: Number},
+    });
+    ds.automigrate(done);
+  });
+
+  describe('model data gets updated properly', function() {
+    var cos1, cos2;
+    before('create entry', function(done) {
+      this.timeout(0);
+
+      COSModel.create({
+        name: 'Conor',
+        age: 23,
+        shopping: ['apple','orange'],
+      })
+      .then(cosmod => {
+        cos1 = cosmod;
+        console.log("COS!, ",cos1);
+
+        return COSModel.create({
+          name: 'John',
+          age: 51,
+          shopping: ['chocolate','crisps']
+        })
+      })
+      .then(cosmod2 => {
+        cos2=cosmod2;
+        console.log("COS2 ",cos2);
+        done();
```

Comment:
Don't call `done`, just return the promise instead.

```
before('create entry', function() {
       this.timeout(0);
  
      return COSModel.create({
      //etc.
});
```

The same comment applies to all other tests below.